src/overlays/actors/ovl_En_Dekubaba/z_en_dekubaba.c

```diff
@@ -44,188 +54,1324 @@ const ActorInit En_Dekubaba_InitVars = {
+}
+
+void EnDekubaba_SpawnIceEffects(EnDekubaba* this, PlayState* play) {
+    if (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+        this->collider.base.colType = COLTYPE_HIT6;
+        this->drawDmgEffAlpha = 0.0f;
+        Actor_SpawnIceEffects(play, &this->actor, this->bodyPartsPos, ARRAY_COUNT(this->bodyPartsPos), 4,
+                              this->size * 0.3f, this->size * 0.2f);
+        this->actor.flags |= ACTOR_FLAG_400;
+    }
+}
+
+void EnDekubaba_SetupWait(EnDekubaba* this) {
+    s32 i;
+    ColliderJntSphElement* element;
+
+    this->actor.shape.rot.x = -0x4000;
+    this->stemSectionAngle[0] = this->stemSectionAngle[1] = this->stemSectionAngle[2] = this->actor.shape.rot.x;
+
+    this->actor.world.pos.x = this->actor.home.pos.x;
+    this->actor.world.pos.z = this->actor.home.pos.z;
+    this->actor.world.pos.y = this->actor.home.pos.y + 14.0f * this->size;
+
+    Actor_SetScale(&this->actor, this->size * 0.01f * 0.5f);
+
+    this->collider.base.colType = COLTYPE_HARD;
+    this->collider.base.acFlags |= AC_HARD;
+    this->timer = 45;
+
+    for (i = 1; i < ARRAY_COUNT(this->colliderElements); i++) {
+        element = &this->collider.elements[i];
+        element->dim.worldSphere.center.x = this->actor.world.pos.x;
+        element->dim.worldSphere.center.y = (s32)this->actor.world.pos.y - 7;
+        element->dim.worldSphere.center.z = this->actor.world.pos.z;
+    }
+
+    this->actionFunc = EnDekubaba_Wait;
+}
+
+void EnDekubaba_Wait(EnDekubaba* this, PlayState* play) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    this->actor.world.pos.x = this->actor.home.pos.x;
+    this->actor.world.pos.z = this->actor.home.pos.z;
+    this->actor.world.pos.y = this->actor.home.pos.y + 14.0f * this->size;
+
+    if ((this->timer == 0) && (this->actor.xzDistToPlayer < 200.0f * this->size) &&
+        (fabsf(this->actor.playerHeightRel) < 30.0f * this->size)) {
+        EnDekubaba_SetupGrow(this);
+    }
+}
+
+void EnDekubaba_SetupGrow(EnDekubaba* this) {
+    s32 i;
+
+    Animation_Change(&this->skelAnime, &gDekuBabaFastChompAnim,
+                     Animation_GetLastFrame(&gDekuBabaFastChompAnim) * (1.0f / 15), 0.0f,
+                     Animation_GetLastFrame(&gDekuBabaFastChompAnim), ANIMMODE_ONCE, 0.0f);
+
+    this->timer = 15;
+
+    for (i = 2; i < ARRAY_COUNT(this->colliderElements); i++) {
+        this->collider.elements[i].info.ocElemFlags |= OCELEM_ON;
+    }
+
+    this->collider.base.colType = COLTYPE_HIT6;
+    this->collider.base.acFlags &= ~AC_HARD;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_DEKU_WAKEUP);
+    this->actionFunc = EnDekubaba_Grow;
+}
+
+void EnDekubaba_Grow(EnDekubaba* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    f32 headDistHorizontal;
+    f32 headDistVertical;
+    f32 headShiftX;
+    f32 headShiftZ;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
```

Comment:
Guessing this can't just be `DECR(this->timer)`?

Reply:
No, most of them it causes trouble if you try and use the macro.